core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -187,16 +188,20 @@ private void createZkNamespace() {
     String schemaRegistryNamespace = "/" + schemaRegistryZkNamespace;
     schemaRegistryZkUrl = zkConnForNamespaceCreation + schemaRegistryNamespace;
 
-    ZkClient zkClientForNamespaceCreation = new ZkClient(zkConnForNamespaceCreation,
-                                                         zkSessionTimeoutMs, zkSessionTimeoutMs,
-                                                         new ZkStringSerializer());
+    ZkUtils zkUtilsForNamespaceCreation = ZkUtils.apply(
+        zkConnForNamespaceCreation,
+        zkSessionTimeoutMs, zkSessionTimeoutMs,
+        JaasUtils.isZkSecurityEnabled(System.getProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM)));
     // create the zookeeper namespace using cluster.name if it doesn't already exist
-    ZkUtils.makeSurePersistentPathExists(zkClientForNamespaceCreation, schemaRegistryNamespace);
+    zkUtilsForNamespaceCreation.makeSurePersistentPathExists(
```

Comment:
We could definitely make this optional, but I think the idea was to provide an area in ZK dedicated to schema registry, allow multiple schema registries in a single ZK cluster if needed (and so it needs to be configurable, though we'd generally discourage having more than one unless absolutely necessary), but also avoid some operational complexity in the common case (single schema registry cluster in one data center running against one ZK cluster).

You're actually evaluating some previous code/design decisions that this patch is just updating and which I wasn't involved in, so you may need @granders or @nehanarkhede to help give a more detailed answer.